File: sdk/r/R/virtualizor.R
#' Virtualizor: VPS management via Virtualizor panel
#' @param http HttpClient object
#' @return Resource object
#' @export
VirtualizorResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "VirtualizorResource"
  env
}

#' POST /api/service/{id}/vms/{vmid}/suspend - Suspend virtual server
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
suspend <- function(self, ...) {
  self$http$post("/api/service/{id}/vms/{vmid}/suspend", list(...))
}

#' POST /api/service/{id}/vms/{vmid}/unsuspend - Unsuspend virtual server
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
unsuspend <- function(self, ...) {
  self$http$post("/api/service/{id}/vms/{vmid}/unsuspend", list(...))
}

#' GET /api/service/{id}/vms/{vmid}/rebuild - List rebuild templates
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listRebuildTemplates <- function(self, ...) {
  self$http$get("/api/service/{id}/vms/{vmid}/rebuild", list(...))
}

#' POST /api/service/{id}/vms/{vmid}/addsshkey - Change SSH key
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
changeSshKey <- function(self, ...) {
  self$http$post("/api/service/{id}/vms/{vmid}/addsshkey", list(...))
}

File: sdk/c/src/virtualizor.c
#include "hitechcloud/virtualizor.h"
#include <stdlib.h>


/* POST /api/service/{id}/vms/{vmid}/suspend - Suspend virtual server */
char *hc_virtualizor_suspend(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/vms/{vmid}/suspend", NULL);
}

/* POST /api/service/{id}/vms/{vmid}/unsuspend - Unsuspend virtual server */
char *hc_virtualizor_unsuspend(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/vms/{vmid}/unsuspend", NULL);
}

/* GET /api/service/{id}/vms/{vmid}/rebuild - List rebuild templates */
char *hc_virtualizor_list_rebuild_templates(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/vms/{vmid}/rebuild", NULL);
}

/* POST /api/service/{id}/vms/{vmid}/addsshkey - Change SSH key */
char *hc_virtualizor_change_ssh_key(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/vms/{vmid}/addsshkey", NULL);
}

